### PR TITLE
[ADD] product: add multi-checkbox attribute type for extra options

### PR DESCRIPTION
**Pull request by vava-odoo**
Currently there is no efficient way to suggest extra-options to customers in PoS or eCommerce. Extra options should be easily selected by customers (or PoS waiters) with checkbox. Each checked option would add an extra-price to the product price and appear on tickets, receipts, invoices, kitchen display, ...

This commit introduces a new type of attribute: multi-checkbox. This will not create product variants. A default extra-cost is defined at the attribute level, but it can be changed at the product level.

The new attribute is available in Sales, not yet in eCommerce and PoS (will be in a future task).

task-3256585

### CODE REVIEW

File: addons/website_sale_comparison/models/website_sale_comparison.py [hunk 1]
```diff
@@ -20,9 +20,13 @@ class ProductAttribute(models.Model):
     _inherit = 'product.attribute'
     _order = 'category_id, sequence, id'
 
-    category_id = fields.Many2one('product.attribute.category', string="Category", index=True,
-                                  help="Set a category to regroup similar attributes under "
-                                  "the same section in the Comparison page of eCommerce")
+    category_id = fields.Many2one(
+        'product.attribute.category',
```
Comment (Feyensv) [inline addons/website_sale_comparison/models/website_sale_comparison.py:24]:
```suggestion
        comodel_name='product.attribute.category',
```

**Review by Feyensv — COMMENTED** [review]
:running_man:

File: addons/product/models/product_attribute.py [hunk 2]
```diff
@@ -63,6 +64,19 @@ def _compute_products(self):
...
+        if self.display_type == 'multi' and self.create_variant != 'no_variant':
+            return {
+                'warning': {
+                    'title': _("Warning"),
+                    'message': _("""
+                        Multi-checkbox display type cannot create variants.
+                        Please select 'Never' as creation mode if you want this display type.
+                    """),
+                }
+            }
```
[the remark this answers is not in the document]
Comment (vava-odoo) [reply addons/product/models/product_attribute.py:78]: I agree the current solution is not ideal. But if I recall correctly the discussion we had with dala, the constrain would be raised too late from a UX POV, as the correct combination should be decided before moving on with the attribute values

[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/product/models/product_attribute.py:78]: I don't mind keeping the onchange, but the onchange only gives a warning, we need a constraint to make sure he cannot save the record with conflicting values.

File: addons/product/models/product_template_attribute_value.py [hunk 3]
```diff
@@ -34,7 +34,9 @@ def _get_default_color(self):
     # configuration fields: the price_extra and the exclusion rules
     price_extra = fields.Float(
         string="Value Price Extra",
-        default=0.0,
+        compute="_compute_price_extra",
+        readonly=False,
+        store=True,
```
[the remark this answers is not in the document]
Comment (vava-odoo) [reply addons/product/models/product_template_attribute_value.py:39]: I don't really understand the whole thing of this method, but I don't think I need it as the compute I added is not a *real* one, but another default instead. I make the change.

File: addons/product/views/product_attribute_value_views.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="product_attribute_value_list" model="ir.ui.view">
+        <field name="name">product.attribute.value.tree</field>
```
Comment (Feyensv) [inline addons/product/views/product_attribute_value_views.xml:4]:
```suggestion

    <record id="product_attribute_value_list" model="ir.ui.view">
        <field name="name">product.attribute.value.list</field>
```

File: addons/sale/models/sale_order_line.py [hunk 5]
```diff
@@ -365,12 +365,21 @@ def _get_sale_order_line_multiline_description_variants(self):
 
         custom_ptavs = self.product_custom_attribute_value_ids.custom_product_template_attribute_value_id
         no_variant_ptavs = self.product_no_variant_attribute_value_ids._origin
+        extra_options = no_variant_ptavs.filtered(lambda ptav: ptav.display_type == 'multi').sorted()
```
Comment (Feyensv) [inline addons/sale/models/sale_order_line.py:368]:
```suggestion
        multi_ptavs = no_variant_ptavs.filtered(lambda ptav: ptav.display_type == 'multi').sorted()
```

File: addons/sale_product_configurator/static/src/js/product_configurator_dialog/product_configurator_dialog.js [hunk 6]
```diff
@@ -206,17 +206,30 @@ export class ProductConfiguratorDialog extends Component {
...
+            const ptavID = parseInt(ptavId);
+            if (!selectedIds.includes(ptavID)){
+                selectedIds.push(ptavID);
+            }
+            else {
+                selectedIds = selectedIds.filter(ptav => ptav !== ptavID);
+            }
+
+        }
+        else {
```
Comment (Feyensv) [inline addons/sale_product_configurator/static/src/js/product_configurator_dialog/product_configurator_dialog.js:229]:
```suggestion
        } else {
```

File: addons/sale_product_configurator/static/src/js/product_configurator_dialog/product_configurator_dialog.js [hunk 7]
```diff
@@ -206,17 +206,30 @@ export class ProductConfiguratorDialog extends Component {
...
-        product.attribute_lines.find(
-            ptal => ptal.id === ptalId
-        ).selected_attribute_value_id = parseInt(ptavId);
+        let selectedIds = product.attribute_lines.find(ptal => ptal.id === ptalId).selected_attribute_value_ids;
+        if (multiIdsAllowed) {
+            const ptavID = parseInt(ptavId);
+            if (!selectedIds.includes(ptavID)){
+                selectedIds.push(ptavID);
+            }
+            else {
```
Comment (Feyensv) [inline addons/sale_product_configurator/static/src/js/product_configurator_dialog/product_configurator_dialog.js:224]: else on closing line

**Review by Feyensv — COMMENTED** [review]
nitpickings

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**